pwiz/data/vendor_readers/Bruker/SpectrumList_Bruker.cpp

```diff
@@ -485,6 +490,26 @@ PWIZ_API_DECL SpectrumPtr SpectrumList_Bruker::spectrum(size_t index, DetailLeve
                 }
             }
         }
+
+        if (result->getMZArray() != nullptr)
+        {
+            auto mz = result->getMZArray()->data;
+
+            if (mz.size() > 0)
+            {
+                auto it = std::min_element(mz.begin(), mz.end());
+                std::size_t index1 = std::distance(mz.begin(), it);
+
+                it = std::max_element(mz.begin(), mz.end());
+                std::size_t index2 = std::distance(mz.begin(), it);
+
+                if (mz.size() > index1)
+                    result->set(MS_lowest_observed_m_z, mz[index1], MS_m_z);
+
+                if (mz.size() > index2)
+                    result->set(MS_highest_observed_m_z, mz[index2], MS_m_z);
+            }
+        }

```

Comment:
The FullMetadata detail level should provide the same metadata as the FullData detail level, and this change would violate that (only FullData would provide lowestMz). This is why we have the spectrum list filter MetadataFixer, if the user really wants these values even though they're not actually provided by the API. So revert this please.

Reply:
Sorry, but I was not aware of the Metadatafixer. But that's a good thing, then you don't need my code here. How is the class used? We use msconvert.exe with the appropriate parameters. What would you have to do to get the two parameters (lowest / highest mz) into the result file?